stable_baselines/gail/dataset/dataset.py

```diff
@@ -120,11 +120,32 @@ def init_dataloader(self, batch_size):
     def __del__(self):
         del self.dataloader, self.train_loader, self.val_loader
 
-    def prepare_pickling(self):
+    def __getstate__(self):
         """
-        Exit processes in order to pickle the dataset.
+        Gets state for pickling.
+
+        Excludes processes that are not pickleable
+        """
+        # Remove processes in order to pickle the dataset.
+        excluded = {'dataloader', 'train_loader', 'val_loader'}
+        state = {key:val for key, val in self.__dict__.items() if key not in excluded}
```

Comment:
I think our codebase normally has spaces after `:`

Reply:
true, but I'm not sure for list comprehension... I will check